packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -139,12 +167,71 @@ async function* traverseRoutesConfig(options: {
           compiler,
           parentInjector: injector,
           parentRoute: currentRoutePath,
+          serverConfigRouteTree,
+          invokeGetPrerenderPaths,
         });
       }
     }
   }
 }
 
+/**
+ * Handles SSG (Static Site Generation) routes by invoking `getPrerenderPaths` and yielding
+ * all parameterized paths.
+ *
+ * @param metadata - The metadata associated with the route tree node.
+ * @param parentInjector - The dependency injection container for the parent route.
+ * @param invokeGetPrerenderPaths - A flag indicating whether to invoke the `getPrerenderPaths` function.
+ * @returns An async iterable iterator that yields route tree node metadata for each SSG path.
+ */
+async function* handleSSGRoute(
+  metadata: ServerConfigRouteTreeNodeMetadata,
+  parentInjector: Injector,
+  invokeGetPrerenderPaths: boolean,
+): AsyncIterableIterator<RouteTreeNodeMetadata> {
+  if (metadata.renderMode !== RenderMode.SSG) {
+    return;
+  }
+
+  const { route: currentRoutePath, getPrerenderPaths, fallback, ...meta } = metadata;
+
+  if (invokeGetPrerenderPaths && currentRoutePath.includes(':')) {
+    if (!getPrerenderPaths) {
+      throw new Error(
+        `The Angular route '${currentRoutePath}' includes parameters, but 'getPrerenderPaths' is missing. ` +
+          `Please define 'getPrerenderPaths' in your server routing configuration.`,
+      );
+    }
+
+    const parameters = await runInInjectionContext(parentInjector, () => getPrerenderPaths());
+
+    for (const params of parameters) {
+      const routeWithResolvedParams = currentRoutePath.replace(URL_PARAMETER_REGEXP, (match) => {
+        const parameterName = match.slice(1);
+        const value = params[parameterName];
+        if (value === undefined) {
+          throw new Error(
+            `The 'getPrerenderPaths' function returned an undefined value for parameter '${parameterName}' in route '${currentRoutePath}'.`,
```

Comment:
We can throw an error, but it is worth failings builds for unused parameters?

Reply:
I was mainly thinking this could be indicative of a bug where a user has a route like `/product/:id` and accidentally returns `{ di: 1234 }`. But I think we should already throw if the user doesn't return `id` in that case, so maybe it's not too big a deal? You'd need to add a new parameter and then forget to update the route which is a bit harder mistake to make.

Up to you if it's worth throwing for.